todo.js

```diff
@@ -0,0 +1,112 @@
+import { v4 as uuidv4 } from 'uuid'
+
+export class Todo {
+    #id
+    #isDone
+    #text
+
+    constructor(text) {
+        this.#id = uuidv4()
+        this.#isDone = false
+        this.#text = text
+    }
+
+    getId() {
+        return this.#id
+    }
+
+    getIsDone() {
+        return this.#isDone
+    }
+
+    setIsDone(value) {
+        this.#isDone = value
+    }
+
+    setText(value) {
+        this.#text = value
+    }
+
+    render(onClick, todos, taskName) {
+        const newDiv = document.createElement('div')
+        const newListItem = this.#createNewListItem(onClick)
+        const newSpan = this.#createSpanBtns(todos)
+
+        newDiv.classList.add('list-item')
+        newListItem.classList.add('list-item__li-tag')
+        newListItem.textContent = taskName.value
+        newDiv.append(newListItem)
+        newDiv.append(newSpan)
+
+        return newDiv
+    }
+
+    #createNewListItem(onClick) {
+        const newListItem = document.createElement('li')
+
+        newListItem.addEventListener('click', event => {
+            if (newListItem.textContent !== '') {
+                onClick(event, this.#id)
+            }
+        })
+
+        return newListItem
+    }
+
+    #createSpanBtns(todos) {
+        const span = document.createElement('span')
+        const trashIcon = document.createElement('i')
+        const pencilIcon = document.createElement('i')
+
+        trashIcon.classList.add('fa-solid', 'fa-trash-can')
+        pencilIcon.classList.add('fa-solid', 'fa-pencil')
+
+        trashIcon.addEventListener('click', event => {
+            event.target.parentNode.parentNode.remove()
+            todos = todos.filter(element => element.getId() !== this.#id)
```

Comment:
no, class shouldnt be aware of some external array

Reply:
I don't really get it, cuz problem here is that class Todo represents a single todo task, not a collection of them, so it's seems to be a bit weird to put a list of items inside each of them and then try to store them all in every instance of Todo, includes storing itself. Perhaps, the solution is to create a new class to store todos? Or remove this method from a class?

Reply:
no, you need to keep array of todo instances out of the class. You shouldn't pass here `todos`. Todo class shouldn't know anything about any array. It simply doesnt care